src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,15 +33,24 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={value}
+        onChange={event => handleSearchChange(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
+      {buttonStatus !== false && value.trim() !== '' && (
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+            onClick={clearSearchButton}
+          />
+        </span>
+      )}
```

Comment:
The condition `buttonStatus !== false && value.trim() !== ''` is used to render the clear button. Since `buttonStatus` is a boolean, you can simplify this condition to `buttonStatus && value.trim() !== ''` for clarity.